plutus-tx-plugin/test/Plugin/NoTrace/trace.plc.golden

```diff
@@ -0,0 +1,9 @@
+(program
+  (let
+    (nonrec)
+    (datatypebind
+      (datatype (tyvardecl Unit (type))  Unit_match (vardecl Unit Unit))
+    )
+    (lam ds (con string) Unit)
```

Comment:
The current simplifier could inline and reduce `\(x :: B.BuiltinString) -> B.trace x ()` as it is only used once. I'll attempt a general solution to also cover `traceComplex`.

Reply:
There already is single-usage inlining, so I'm surprised if this isn't happening. I think a plausible heuristic would be to say something like "inline lambdas whose bodies are trivial".

Reply:
It already is happening! This was a note saying that this test was inlined and reduced because of its single usage, not because of trivial bodies or unused arguments, which a future PR should attempt, and test on the not-inlined `traceComplex` case.

Reply:
Ah right, I see